Sources/tart/VM.swift

```diff
@@ -91,12 +91,17 @@ class VM: NSObject, VZVirtualMachineDelegate, ObservableObject {
     // Download the IPSW
     defaultLogger.appendNewLine("Fetching \(remoteURL.lastPathComponent)...")
 
-    let (channel, response) = try await Fetcher.fetch(URLRequest(url: remoteURL), viaFile: true)
+    let downloadProgress = Progress(totalUnitCount: 100)
+    ProgressObserver(downloadProgress).log(defaultLogger)
 
-    let progress = Progress(totalUnitCount: response.expectedContentLength)
-    ProgressObserver(progress).log(defaultLogger)
+    let request = URLRequest(url: remoteURL)
+    let (channel, response) = try await Fetcher.fetch(request, viaFile: true, progress: downloadProgress)
 
     let temporaryLocation = try Config().tartTmpDir.appendingPathComponent(UUID().uuidString + ".ipsw")
+    defaultLogger.appendNewLine("Computing digest for \(temporaryLocation.path)...")
+    let digestProgress = Progress(totalUnitCount: response.expectedContentLength)
+    ProgressObserver(digestProgress).log(defaultLogger)
+
```

Comment:
Why do compute the digest here? We already have the hash from `x-amz-meta-digest-sha256` above. If we don't trust that the download actually delivers a file with the same hash we should probably compare the hashes, but I don't think we need to be that paranoid do we? 

Copying the data via the loop below takes a bit of time, compared to just moving the file, so perhaps we can just move the file from the `URLSession` temporary location into the Tart IPSW cache?

Reply:
> Why do compute the digest here? We already have the hash from `x-amz-meta-digest-sha256` above.

Because `--from-ipsw` might point to an arbitrary URL, which will not always be an S3 storage.

Reply:
Ah, of course. Since the typical use-case is pulling from Apple's S3 storage, would it make sense to skip the digest for those cases we do have a pre-computed hash and just move the file directly? Would speed up downloads by 30ish seconds in my local testing (even the copy takes some time, even without the digest).

Reply:
Anyways, this is something to consider beyond this PR, so I'll close this conversation as "resolved" to mark this PR as "complete" :)